src/Tools/dotnet-monitor/CollectionRules/Actions/LoadProfilerAction.cs

```diff
@@ -62,6 +62,8 @@ public LoadProfilerAction(ILogger logger, IEndpointInfo endpointInfo, LoadProfil
                 {
                     DiagnosticsClient client = new DiagnosticsClient(_endpointInfo.Endpoint);
 
+                    await client.SetEnvironmentVariableAsync("DotnetMonitorProfiler_InstanceId", _endpointInfo.RuntimeInstanceCookie.ToString("D"), token);
```

Comment:
I don't think this should be here. This is very specific to the monitor profiler whereas the LoadProfilerAction is mean for general usage.

Reply:
Will use pid for now. We will likely need to add a token for runtime instance id to add to the set environment action.